custom_components/unifiprotect/sensor.py

```diff
@@ -157,25 +297,62 @@ def __init__(
         self,
         protect: ProtectApiClient,
         protect_data: UnifiProtectData,
-        device: ProtectAdoptableDeviceModel,
+        device: ProtectDeviceModel,
         description: UnifiProtectSensorEntityDescription,
     ):
         """Initialize an Unifi Protect sensor."""
         self.entity_description: UnifiProtectSensorEntityDescription = description
         super().__init__(protect, protect_data, device, description)
         self._attr_name = f"{self.device.name} {self.entity_description.name}"
 
-    @property
-    def native_value(self) -> StateType | date | datetime:
-        """Return the state of the sensor."""
-        value = get_nested_attr(self.device, self.entity_description.ufp_value)
-
-        assert isinstance(value, (str, int, float, datetime)) or value is None
-        if isinstance(value, datetime) and not above_ha_version(2021, 12):
-            return value.replace(microsecond=0).isoformat()
+    @callback
+    def _async_nvr_value(self) -> float:
+        assert isinstance(self.device, NVR)
+        if self.entity_description.key == _KEY_MEMORY:
+            memory = self.device.system_info.memory
+            value = (1 - memory.available / memory.total) * 100
```

Comment:
```suggestion
        if self.entity_description.key == _KEY_MEMORY:
            memory = self.device.system_info.memory
            return (1 - memory.available / memory.total) * 100
```
Drop the `else`
Then outdent below.

Reply:
I actually do not need like any of that anymore. That was the old code before I make the pyunifiprotect changes.